custom_components/google_home/switch.py

```diff
@@ -0,0 +1,87 @@
+"""Switch platform for Google Home"""
+import logging
+from typing import Any, Callable, Iterable, List
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    DATA_CLIENT,
+    DATA_COORDINATOR,
+    DOMAIN,
+    ICON_DO_NOT_DISTURB,
+    LABEL_DO_NOT_DISTURB,
+)
+from .entity import GoogleHomeBaseEntity
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_devices: Callable[[Iterable[SwitchEntity]], None],
+) -> bool:
+    """Setup switch platform."""
+    client = hass.data[DOMAIN][entry.entry_id][DATA_CLIENT]
+    coordinator = hass.data[DOMAIN][entry.entry_id][DATA_COORDINATOR]
+
+    switch: List[SwitchEntity] = []
+    for device in coordinator.data:
+        if device.auth_token and device.available:
+            switch.append(
+                DoNotDisturbSwitch(
+                    coordinator,
+                    client,
+                    device.name,
+                )
+            )
+
+    if switch:
+        async_add_devices(switch)
+
+    return True
+
+
+class DoNotDisturbSwitch(GoogleHomeBaseEntity, SwitchEntity):
+    """Google Home Do Not Disturb switch."""
+
+    @property
+    def label(self) -> str:
+        """Label to use for name and unique id."""
+        return LABEL_DO_NOT_DISTURB
+
+    @property
+    def icon(self) -> str:
+        """Return the icon of the sensor."""
+        return ICON_DO_NOT_DISTURB
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if switch is on."""
+        device = self.get_device()
+
+        if device is None:
+            return False
+
+        is_enabled = device.get_do_not_disturb()
+
+        return not is_enabled
```

Comment:
I think it should be the opposite. The switch is on when do not disturb mode is on.
```suggestion
        return device.get_do_not_disturb()
```

Reply:
That is how it is now. The setting on device endpoint is inverted. Or do you mean that you want the switch to be off when do not disturb is on? 🤔

Reply:
No, the switch should be on, when do not disturb is on.

`device.get_do_not_disturb()` should return `True` when it's on.

This is how you set it, you set `True` to turn on and `False` to turn off.